packages/cli/src/components/PreviewSender.tsx

```diff
@@ -12,19 +12,24 @@ const PreviewSender: React.FC<PreviewSenderProps> = ({
   previewClass,
 }) => {
   const [email, setEmail] = useState<string | null>(null);
+  const [usedEmailFromLocalStorage, setUsedEmailFromLocalStorage] =
+    useState<boolean>(false);
   const [error, setError] = useState<React.ReactElement | null>(null);
   const [lastSendAt, setLastSentAt] = useState<Date | null>(null);
   const [sending, setSending] = useState<boolean>(false);
 
   useEffect(() => {
-    if (!email) {
+    if (!email && !usedEmailFromLocalStorage) {
       setEmail(window.localStorage.getItem("previewSenderEmail"));
       const lastSent = window.localStorage.getItem("previewSenderLastSentAt");
       if (lastSent) {
         setLastSentAt(new Date(lastSent));
       }
+
+      // ensure this only happens once
+      setUsedEmailFromLocalStorage(true);
     }
-  }, [email]);
+  }, [email, usedEmailFromLocalStorage]);
```

Comment:
There's a way to do this with less code by passing [] (https://reactjs.org/docs/hooks-effect.html), but I would have to disable the tslint for this line:  "React Hook useEffect has missing dependencies: 'email' and 'usedEmailFromLocalStorage'. Either include them or remove the dependency array."

> If you want to run an effect and clean it up only once (on mount and unmount), you can pass an empty array ([]) as a second argument. This tells React that your effect doesn’t depend on any values from props or state, so it never needs to re-run. This isn’t handled as a special case — it follows directly from how the dependencies array always works.

wdyt?

Reply:
would this work?

```
  useEffect(() => {
       setEmail(window.localStorage.getItem("previewSenderEmail"));
       const lastSent = window.localStorage.getItem("previewSenderLastSentAt");
       if (lastSent) {
         setLastSentAt(new Date(lastSent));
       }
   }, []);
```

Reply:
yes, but ts lint does not allow